test(linter/no-unused-vars): ignored catch parameters

DonIsaac opened the pull request:
Add test cases covering unused/ignored catch parameters after [this discussion](https://discord.com/channels/1079625926024900739/1080712072012238858/1295014826388623414) on discord.

DonIsaac commented:
* **#6558** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6558?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#6557** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6557?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#6556** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6556?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#6555** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6555?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @DonIsaac and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

camc314 reviewed (approved):
should delete https://oxc.rs/docs/guide/usage/linter/rules/unicorn/prefer-optional-catch-binding.html as i think this covers it?

DonIsaac commented:
> should delete https://oxc.rs/docs/guide/usage/linter/rules/unicorn/prefer-optional-catch-binding.html as i think this covers it?

IMO we should leave it, in case people want to enable that rule but disable `no-unused-vars`

DonIsaac committed:
test(linter/no-unused-vars): ignored catch parameters (#6555)

Add test cases covering unused/ignored catch parameters after [this discussion](https://discord.com/channels/1079625926024900739/1080712072012238858/1295014826388623414) on discord.
